src/Libraries/Microsoft.Extensions.AI.Ollama/OllamaChatClient.cs

```diff
@@ -271,7 +272,10 @@ private static ChatMessage FromOllamaMessage(OllamaChatResponseMessage message)
             contents.Insert(0, new TextContent(message.Content));
         }
 
-        return new ChatMessage(new(message.Role), contents);
+        // Ollama doesn't set a response ID on responses or messages, so generate one for consistency with streaming
+        var messageId = Guid.NewGuid().ToString("N");
+
+        return new ChatMessage(new(message.Role), contents) { MessageId = messageId };
```

Comment:
Is this message ID then inherited to be the response ID as well?

Trying to rationalize this with the AzureAIInference one above, where it sets them to be the same.

Reply:
Good point. This was inconsistent. I've pushed an update to align it.